### ISSUE TITLE
Get the artist genres and expose it as a template variable

### ISSUE THREAD

**Issue by elvarb** [studiowebux/obsidian-spotify-link#17]
This would require an additional request using the Get Artist API only to be used if the template has a genre variable.

https://developer.spotify.com/documentation/web-api/reference/get-an-artist

> "genres": ["dance pop", "miami hip hop", "pop"],

### 1. {{genres}} would return the string values from the array

> dance pop, miami hip hop, pop

### 2. {{genres_array}} would return the raw array to be used as a file or inline property

> "dance pop", "miami hip hop", "pop"

Inline usage:

> Genres:: "dance pop", "miami hip hop", "pop"

### 3. {{genres_hashtag}} would return the values as hashtags

> #dance_pop #hip_hop #pop

## Other potential uses from the artist api

- Follower count
- Popularity value
- Included is also the Artist image, which sometimes is more desirable to have instead of the album art.

**Comment by studiowebux** [comment]
Hello @elvarb 

I did a first pass focusing on the genres and implemented the artist API 

**Input**:

```md
Genres Tests:

Genres: {{ genres }}
Genres Array: {{ genres_array }}
Genres hashatg: {{ genres_hashtag }}
```

**Output:**

```md
Genres Tests:

Genres: modern rock, permanent wave, punk, rock
Genres Array: "modern rock", "permanent wave", "punk", "rock"
Genres hashatg: #modern_rock #permanent_wave #punk #rock
```

Does it work for you ?

**Comment by elvarb** [comment]
That is excellent, now this plugin gives me more information than Spotify does 👍

**Comment by studiowebux** [comment]
Hello @elvarb 

I released v1.6.0 (I will have a small fix to release for the setting documentation, but has no functional impact)

The summary:

- Added `artist_name`, `popularity`, `followers`, `artist_image`, `genres`, `genres_array`, `genres_hashtag`
- If only one artist if defined it prints a unique value, otherwise it prefix the artist name followed by the value
